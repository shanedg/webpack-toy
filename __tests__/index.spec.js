import ClaimLib from '../src/lib';

describe('AoC \'18 - Day 3', () => {
  const originalLog = console.log;
  afterEach(() => console.log = originalLog);

  describe('check answers printed to stdout', () => {
    const consoleOutput = [];
    const mockedLog = output => consoleOutput.push(output);
    beforeEach(() => console.log = mockedLog);

    test('snapshot correct results', async () => {
      await new ClaimLib().run();
      expect(consoleOutput.length).toBeGreaterThan(0);
      expect(JSON.stringify(consoleOutput)).toMatchSnapshot();
    });
  });

});
