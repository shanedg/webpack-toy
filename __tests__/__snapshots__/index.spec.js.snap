// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AoC '18 - Day 3 check answers printed to stdout snapshot correct results 1`] = `"[\\"common squ in: 107820\\",\\"claims w/o overlap:\\\\n[{\\\\\\"id\\\\\\":661,\\\\\\"x\\\\\\":785,\\\\\\"y\\\\\\":105,\\\\\\"width\\\\\\":27,\\\\\\"height\\\\\\":15}]\\"]"`;
